lib/promiscuous/subscriber/worker/distributor.rb

```diff
@@ -0,0 +1,88 @@
+class Promiscuous::Subscriber::Worker::Distributor
+  def initialize(root)
+    @root = root
+  end
+
+  def start
+    num_threads = Promiscuous::Config.subscriber_threads
+    Promiscuous::Config.subscriber_topics.each do |topic|
+      @distributor_threads ||= num_threads.times.map { DistributorThread.new(topic) }
+      Promiscuous.debug "[distributor] Started #{num_threads} thread#{'s' if num_threads>1} topic:#{topic}"
+    end
+  end
+
+  def stop
+    return unless @distributor_threads
+    Promiscuous.debug "[distributor] Stopping #{@distributor_threads.count} threads"
+
+    @distributor_threads.each { |distributor_thread| distributor_thread.stop }
+    @distributor_threads = nil
+  end
+
+  def show_stop_status(num_requests)
+    @distributor_threads.to_a.each { |distributor_thread| distributor_thread.show_stop_status(num_requests) }
+  end
+
+  class DistributorThread
+    def initialize(topic)
+      # late include of CelluloidSubscriber because the class is resolved
```

Comment:
haha thanks for the comment -- it took me and @TomHoss a while to understand why another extend in promiscuous was inside initialize